valhalla/sif/dynamiccost.h

```diff
@@ -307,6 +341,7 @@ class DynamicCost {
         // Ids do not match the path for this restriction.
         bool match = true;
         const EdgeLabel* next_pred = first_pred;
+        std::list<baldr::GraphId> edge_ids_in_complex_restriction;
```

Comment:
since we only `push_back` we could use a vector and even reserve to match the length

Reply:
We could use a vector, but could it be reserved up front? Do we not actually have to walk the restriction to know how long it is?